File: backend/models/jadwal.go
package models

type Jadwal struct {
	JadwalID    	string `json:"id_jadwal"`
	KodeMK			string `json:"kode_mk"`
	NamaMK      	string `json:"nama_mk"`
	KodeRuangan 	string `json:"kode_ruangan"`
	Hari        	string `json:"hari"`
	JamMulai    	string `json:"jam_mulai"`
	JamSelesai  	string `json:"jam_selesai"`
	DosenPengampu []string `json:"dosen_pengampu"`
	Kelas			string `json:"kelas"`
	SKS         	int    `json:"sks"`
}

File: frontend/siris/lib/class/Jadwal.dart
class Jadwal {
  final int idJadwal;
  final String KodeMK;
  final String NamaMK;
  final String Ruangan;
  final String Hari;
  final String JamMulai;
  final String JamSelesai;
  final String Kelas;
  final int SKS;
  final List<String> DosenPengampu;

  Jadwal({
    required this.idJadwal,
    required this.KodeMK,
    required this.NamaMK,
    required this.Ruangan,
    required this.Hari,
    required this.JamMulai,
    required this.JamSelesai,
    required this.Kelas,
    required this.SKS,
    required this.DosenPengampu
  });

  factory Jadwal.fromJson(Map<String, dynamic> json) {
    return Jadwal(
      idJadwal: json['id_jadwal'],
      KodeMK: json['kode_mk'],
      NamaMK: json['nama_mk'],
      Ruangan: json['kode_ruangan'],
      Hari: json['hari'],
      JamMulai: json['jam_mulai'],
      JamSelesai: json['jam_selesai'],
      Kelas: json['kelas'],
      SKS: json['sks'],
      DosenPengampu: List<String>.from(json['dosen_pengampu']),
    );
  }
}

